src/hotspot/share/gc/z/zPageAllocator.cpp

```diff
@@ -2191,6 +2193,12 @@ bool ZPageAllocator::is_multi_partition_enabled() const {
   return _virtual.is_multi_partition_enabled();
 }
 
+bool ZPageAllocator::is_multi_partition_allowed(const ZPageAllocation* allocation) const {
+  return allocation->type() == ZPageType::large &&
+         is_multi_partition_enabled() &&
+         sum_available() >= allocation->size();
+}
```

Comment:
Thanks for addressing my comments, if I'm being picky I would prefer this order of the statements. If you prefer the current I'm ok with this as well.

```suggestion
  return is_multi_partition_enabled() &&
         allocation->type() == ZPageType::large &&
         allocation->size() <= sum_available();
}
```